board/app/models/thread.php

```diff
@@ -1,122 +1,180 @@
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
 
-            if ($this->hasError() || $comment->hasError())
-            {
-                throw new ValidationException('Invalid thread or comment');
-            }
+    public function isUserThread()     
+    {
+        return $this->user_id === $_SESSION['user_id'];
+    }
 
-        $date_created = date("Y-m-d H:i:s");
-        $params = array(
-            'title' => $this->title,
-            'created'=> $date_created
-             );
-
-    //Latest inserted ID
-            try 
-            {
-                $db = DB::conn();
-                $db->begin();
-                $db->insert('thread', $params);
-                $this->id = $db->lastInsertId();
-
-    //write first comment at the same time
-                $this->write($comment);
-                $db->commit();
-            } catch (Exception $e) {
-                $db->rollback();
-            }
+    public function deleteThread($user_id) 
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                $this->id,
+                $this->user_id
+            );
+            $db->query('DELETE FROM thread WHERE id = ? AND user_id = ?', $params);
+            $db->commit();
+            Comment::deleteAllComments($this->id);
+        } catch (Exception $e) {
+            $db->rollback();
+        }
     }
 
-    public static function getAll($offset, $limit) 
+    public static function getAll($offset, $limit)
     {
         $threads = array();
         $db = DB::conn();
-        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
 
-        foreach($rows as $row) 
-        {
-            $threads[] = new self($row);
+        if (!is_int($offset) || !is_int($limit)) { 
+            throw new NotIntegerException; 
         }
 
-    return $threads;
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
+    
+        foreach($rows as $row) {
+            $threads[] = new self($row);
+        }
+        return $threads;
     }
 
-    public static function countAll()
+    public static function countAll() 
     {
         $db = DB::conn();
         return (int) $db->value('SELECT COUNT(*) FROM thread');
     }
 
-    public function countComments()
+    public static function countAllThreadByUserId($user_id) 
     {
         $db = DB::conn();
-        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ? ", array($this->id));
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($user_id));
     }
 
-
-    public static function get($id)
+    public static function countAllThreadByCategory($category)
+    {
+        $db = DB::conn();
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($category));
```

Comment:
The function name seems to be getting a $category, but in this sql query, you are passing its value to a user_id? :flushed: 
```
user_id = category
```
:astonished:  I think you can improve on the logic of this one po :grin: :bow: